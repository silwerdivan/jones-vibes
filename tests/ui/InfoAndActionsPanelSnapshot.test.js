import { JSDOM } from 'jsdom';
import fs from 'fs';
import path from 'path';

describe('Info and Actions Panel Snapshot Test', () => {
  let dom;

  beforeAll(() => {
    const html = fs.readFileSync(path.resolve(__dirname, '../../index.html'), 'utf8');
    dom = new JSDOM(html);
  });

  it('should match the snapshot of the location info panel', () => {
    const locationInfoPanel = dom.window.document.querySelector('#location-info');
    expect(locationInfoPanel.innerHTML).toMatchSnapshot();
  });

  it('should match the snapshot of the game controls panel', () => {
    const gameControlsPanel = dom.window.document.querySelector('#game-controls');
    expect(gameControlsPanel.innerHTML).toMatchSnapshot();
  });
});
