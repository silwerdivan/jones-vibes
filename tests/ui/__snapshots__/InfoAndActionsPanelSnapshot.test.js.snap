// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Info and Actions Panel Snapshot Test should match the snapshot of the game controls panel 1`] = `
"
                <h2>Actions</h2>
                <div class="action-buttons">
                    <button id="btn-rest-end-turn">Rest / End Turn</button>
                    <button id="btn-travel" class="location-specific hidden">Travel</button>
                    <button id="btn-work-shift" class="location-specific hidden">Work Shift</button>
                    <button id="btn-take-course" class="location-specific hidden">Take Course</button>
                    <button id="btn-buy-item" class="location-specific hidden">Buy Item</button>
                    <button id="btn-deposit" class="location-specific hidden">Deposit</button>
                    <button id="btn-withdraw" class="location-specific hidden">Withdraw</button>
                    <button id="btn-take-loan" class="location-specific hidden">Take Loan</button>
                    <button id="btn-repay-loan" class="location-specific hidden">Repay Loan</button>
                    <button id="btn-buy-car" class="location-specific hidden">Buy Car</button>
                </div>
            "
`;

exports[`Info and Actions Panel Snapshot Test should match the snapshot of the location info panel 1`] = `
"
                <h2>Current Location</h2>
                <p>Current Location: <span id="current-location">Start</span></p>
                <p>Turn: <span id="game-turn">1</span></p>
            "
`;
